scriptworker/task.py

```diff
@@ -250,6 +250,29 @@ def guess_content_type(path):
                     )
 
 
+def _compress_if_text_plain(artifact_path, content_type):
+    if content_type == 'text/plain':
+        log.debug('"{}" is a plain text file. Compressing...'.format(artifact_path))
+        with open(artifact_path, 'rb') as f_in:
+            text_content = f_in.read()
+
+        with gzip.open(artifact_path, 'wb') as f_out:
+            f_out.write(text_content)
+        log.info('Compressed "{}" with gzip'.format(artifact_path))
+    else:
+        log.debug('"{}" is NOT a plain text file, nothing to compress.'.format(artifact_path))
+
+
+def _craft_artifact_put_hearders(tc_response):
+    headers = {
+        aiohttp.hdrs.CONTENT_TYPE: tc_response['contentType'],
+    }
+    if tc_response['contentType'] == 'text/plain':
```

Comment:
same here.  Instead of changing the `contentType` here for all `text/plain`, we could go back to `guess_content_type` where we set the default contentType for all `.log` files to `text/plain`, and change that?  If we decide we want to gzip all `text/plain` artifacts no matter what, then we can keep this.